llvm/test/TableGen/directive1.td

```diff
@@ -342,4 +359,22 @@ def TDL_DirA : Directive<"dira"> {
 // IMPL-NEXT:    llvm_unreachable("Invalid Tdl Directive kind");
 // IMPL-NEXT:  }
 // IMPL-EMPTY:
+// IMPL-NEXT:  llvm::ArrayRef<llvm::tdl::Directive> llvm::tdl::getLeafConstructs(llvm::tdl::Directive Dir) {
+// IMPL-NEXT:    static llvm::ArrayRef<llvm::tdl::Directive> nothing {};
```

Comment:
Is this supposed to be a container for the leaves? ArrayRef is not owning container, I'm afraid it may cause some UB. Better just to use POD arrays here, if I understand correctly generated code.

Reply:
Eh, sorry, it's a mistake.  Thanks for the catch.

Reply:
Returning an empty ArrayRef by value now.  I can't use regular array (as in the case of non-empty sets), because I can't create a zero-length array.  The empty ArrayRef does not refer to any storage, so returning it by value should be ok.